tests/tests_C2_economic_functions.py

```diff
@@ -0,0 +1,131 @@
+import src.C2_economic_functions as e_functions
+
+project_life = 20
+wacc = 0.1
+investment_t0 = 220000
+tax = 0.15
+# please do not change project_life and lifetime as this will affect CAPEX calculations that depend on the number of investments
+lifetime = {
+    "equal project life": project_life,
+    "smaller project life": project_life - 5,
+    "bigger project life": project_life + 15,
+}
+present_value = 295000
+crf = 0.12
+annuity = 35400
+annuity_factor = 1 / 0.12
+fuel_keys = {
+    "fuel_price": 1.3,
+    "fuel_price_change_annual": 0,
+    "project_lifetime": project_life,
+    "wacc": wacc,
+}
+
+
+def test_annuity_factor():
+    """
+
+    Tests whether the MVS is correctly calculating the annuity factor
+    """
+    AF = e_functions.annuity_factor(project_life, wacc)
+    assert AF == 1 / wacc - 1 / (wacc * (1 + wacc) ** project_life)
+
+
+def test_crf():
+    """
+
+    Tests whether the MVS is correctly calculating the capital recovery factor
+    """
+    CRF = e_functions.crf(project_life, wacc)
+    assert CRF == (wacc * (1 + wacc) ** project_life) / ((1 + wacc) ** project_life - 1)
+
+
+def test_capex_from_investment_lifetime_equals_project_life():
+    """
+
+    Tests whether the MVS is correctly calculating the capital expenditure of the project if the lifetime is equal to project_life
+    """
+    CAPEX = e_functions.capex_from_investment(
+        investment_t0, lifetime["equal project life"], project_life, wacc, tax
+    )
+    assert CAPEX == investment_t0 * (1 + tax)
+
+
+def test_capex_from_investment_lifetime_smaller_than_project_life():
```

Comment:
I wonder whether there's a way to shorten the expected value. 
My first suggestion is to define the expected value before the assertion `exp_capex = ...` and then `assert CAPEX == exp_capex`.
I don't know whether my second suggestion helps here: I usually try to use easy numbers so that I can calculate the expected value by heart. This might not account for these economic functions here.. just check whether it's possible.

Reply:
I totally agree with you. However, for the two cases in which lifetime is not equal to project_life, the capex is a float. What precision do you require in that case?

Reply:
I think I don't know what you are asking for.. Do you mean how many decimals behind the dot/comma? 
If you have many decimals you could look into [pytest.approx](https://docs.pytest.org/en/latest/reference.html#pytest-approx).

Reply:
yes that was helpful, thanks :)